clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/artifact/Replacer.java

```diff
@@ -109,6 +112,14 @@ private Replacer(
     return replacedArtifacts.build();
   }
 
+  private Predicate createReplaceFilterPredicate(String replacePath, String name) {
+    return ctx -> {
+      ValueNode node = ValueNode.toValueNode("@." + replacePath).asPathNode().evaluate(ctx);
+      String value = node.asStringNode().getString();
```

Comment:
We should check `node.isStringNode()` before calling `node.asStringNode()` to avoid propagating an exception.  (I think in that case we'd just want to return `false` early as the node didn't have what we expected.)

Reply:
Updated!

Reply:
You're checking `node.isStringNode()` after already calling `node.asStringNode()` by which time the exception will already have happened---you'll need to move the `node.isStringNode()` to happen before we call `node.asStringNode()`.

Reply:
Sorry about that, fixed now.